plugins/ArchivingMetrics/Timer.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\ArchivingMetrics;
+
+use Piwik\ArchiveProcessor\Rules;
+use Piwik\Plugins\ArchivingMetrics\Clock\Clock;
+use Piwik\Plugins\ArchivingMetrics\Clock\ClockInterface;
+use Piwik\Plugins\ArchivingMetrics\Writer\DbWriter;
+use Piwik\Plugins\ArchivingMetrics\Writer\WriterInterface;
+
+final class Timer
+{
+    /**
+     * @var bool
+     */
+    private $isArchivePhpTriggered;
+
+    /**
+     * @var ClockInterface
+     */
+    private $clock;
+
+    /**
+     * @var WriterInterface
+     */
+    private $writer;
+
+    private $runs = [];
+
+    /**
+     * @var Timer
+     */
+    private static $instance;
+
+    public function __construct(bool $isArchivePhpTriggered, ClockInterface $clock, WriterInterface $writer)
+    {
+        $this->isArchivePhpTriggered = $isArchivePhpTriggered;
+        $this->clock = $clock;
+        $this->writer = $writer;
+    }
+
+    public static function getInstance(bool $isArchivePhpTriggered, ?ClockInterface $clock = null, ?WriterInterface $writer = null): self
+    {
+        if (self::$instance !== null) {
+            return self::$instance;
+        }
+
+        if ($clock === null) {
+            $clock = new Clock();
+        }
+
+        if ($writer === null) {
+            $writer = new DbWriter();
+        }
+
+        self::$instance = new self($isArchivePhpTriggered, $clock, $writer);
+        return self::$instance;
+    }
+
+    public function start(Context $context): void
+    {
+        if (false === $this->isApplicableForTiming($context)) {
+            return;
+        }
+
+        $this->runs[$context->getKey()] = [
+            'idsite' => $context->idSite,
+            'period' => $context->period->getLabel(),
+            'segment' => $context->segment->getString(),
+            'date1' => $context->period->getDateTimeStart()->toString('Y-m-d'),
+            'date2' => $context->period->getDateTimeEnd()->toString('Y-m-d'),
+            'ts_started' => $this->clock->now(),
+            'timeStarted' => $this->clock->microtime(),
```

Comment:
Wouldn't it be simpler to just have one value for the starting time, and use that one to later generate the `ts_tarted` inserted into the database?